### PR TITLE
Install protoc to avoid having to recompile it

### PR DESCRIPTION
**Pull request by agourlay**
`prost-build` is the longest compiling dependency we have according to `cargo --timings`.

This is due to `prost` compiling `protocolbuffers` as well during the build https://github.com/tokio-rs/prost/pull/610

Tokio recommends to install the `protoc` binaries to optimize the compilation process https://github.com/tokio-rs/prost#protoc=

This PR adds `sudo apt install protobuf-compiler` to our CI jobs.

I tested this locally and it does work nicely.

### CODE REVIEW

**Comment by agourlay** [comment]
Lots of errors I need to look into :sweat:

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -24,6 +24,9 @@ jobs:
       - uses: Swatinem/rust-cache@v1
       - name: Install dependencies
         run: sudo apt-get install clang
+      - name: Install protoc
+        run: |
+          .github/workflows/protoc.sh
```
Comment (e-ivkov) [inline .github/workflows/integration-tests.yml:29]:
```suggestion
        run: .github/workflows/protoc.sh
```
Same below

**Comment by agourlay** [comment]
Alright I finally got this to work :partying_face: 

The original solution regarding installing `protocolbuffers` via apt did not work on github actions because `ubuntu-latest` means `20.04` which is using an old version of `protoc`.

The solution is to manually install the functioning version found in ubuntu `21.04` which is `3.12.4`.

Regarding the docker container it is currently broken on master with:

```
  --- stderr
  thread 'main' panicked at '
  failed to execute command: No such file or directory (os error 2)
  is `cmake` not installed?

```

This is also a side effect of the tonic upgrade https://github.com/hyperium/tonic/issues/965

As the docker container is not build very often I decided to simply install `cmake`.

The bottom line is that every developer should install a recent `protocolbuffers` locally to avoid wasting time during cold compilation.

**Review by e-ivkov — APPROVED** [review]

**Comment by agourlay** [comment]
## Before
before-protoc

## After
after-protoc

**Review by generall — APPROVED** [review]
